src/GafferScene/SceneAlgo.cpp

```diff
@@ -511,6 +511,12 @@ SceneAlgo::History::Ptr SceneAlgo::history( const Gaffer::ValuePlug *scenePlugCh
 		Monitor::Scope monitorScope( &monitor );
 		scenePlugChild->hash();
 	}
+
+	if( monitor.rootProcesses().size() == 0 )
+	{
+		return nullptr;
```

Comment:
Are you sure this is the most useful behaviour? I presume that because you haven't updated the call site in TransformTool or even the others in this file, we now have more bugs than we started with. Perhaps requiring all callers to make null checks isn't the best approach?

Another option might be to return `new History( scenePlugChild->parent(), new Context( Context::current() ) )`, and to make sure that the usual code path also uses an equivalent item for the root of the history. Currently we define an item of history as plugs where computes take place, and this would deviate from that, but it seems like it might not be surprising to always have the root item represent the arguments to the function? There could even be an argument for including non-computed history items, where a connection simply carries the value through. I'm undecided...

Reply:
> Are you sure this is the most useful behaviour? I presume that because you haven't updated the call site in TransformTool or even the others in this file, we now have more bugs than we started with. Perhaps requiring all callers to make null checks isn't the best approach?

I don't believe there would be any new bugs, as whenever this condition is reached in the current code, we segfault. 
Lets figure out a semantic that makes sense long term, this immediate commit was just to turn a segfault into something less terminal, given its bound to python.

Reply:
After a quick discussion about this, we agreed that we should scope this change to just avoiding the segfault and updating the call sites in Gaffer to handle the `nullptr`.  Longer-term there are thoughts around enabling the history API to support queries that list all nodes that have been traversed (not just those that affected the value).